examples/cifar10_cnn_capsule.py

```diff
@@ -231,6 +216,7 @@ def compute_output_shape(self, input_shape):
     # Fit the model on the batches generated by datagen.flow().
     model.fit_generator(
         datagen.flow(x_train, y_train, batch_size=batch_size),
+        steps_per_epoch= len(x_train) / batch_size,
```

Comment:
The bug you encountered is a bug appearing only in keras 2.2.4. The version from master can run this script without any issues.

Reply:
It seems that on updating to master the script fails at the batch_dot operation.
```python 'y.shape[%d] (%d != %d).' % (axes[0], axes[1], d1, d2))
ValueError: Can not do batch_dot on inputs with shapes (None, 10, 10, 16) and (None, 10, None, 16) with axes=[2, 3]. x.shape[2] != y.shape[3] (10 != 16).
```

Reply:
@farizrahman4u could this be related to tour recent modifications of batch dot?